src/environmentd/src/http/webhook.rs

```diff
@@ -124,12 +126,30 @@ fn pack_row(
     }
 
     // Pack the headers into our row, if required.
-    if header_ty.is_some() {
-        packer.push_dict(
-            headers
-                .iter()
-                .map(|(name, val)| (name.as_str(), Datum::String(val))),
-        );
+    match header_tys {
+        Some(include_headers) if include_headers.selective_headers.is_empty() => {
+            packer.push_dict(
```

Comment:
It is real odd that all headers get included as a dict, whereas named headers as a string. Can we choose one and always use it? Otherwise the validation methods have to differ. Users switching between them will get confused and it'll make the docs weirder.

Reply:
Chatted in [Slack](https://materializeinc.slack.com/archives/C02FWJ94HME/p1693334482449209).

tl;dr to map a header to a column name we'll support `INCLUDE HEADER <header-name> AS <column-name>`. To include all headers as a map and possibly filter them we support `INCLUDE HEADERS ( [NOT] <header-name> )`